fendermint/vm/topdown/src/sync/syncer.rs

```diff
@@ -184,48 +218,24 @@ where
             return Err(Error::ParentChainReorgDetected);
         }
 
-        if let Some((to_confirm_height, to_confirm_hash)) = self.sync_pointers.tail() {
-            tracing::debug!(
+        let data = self.fetch_data(height, block_hash_res.block_hash).await?;
+        atomically_or_err::<_, Error, _>(|| {
+            self.provider.new_parent_view(height, Some(data.clone()))?;
+            emit!(
+                EventType::NewParentView,
+                is_null = false,
                 height,
-                confirm = to_confirm_height,
-                "non-null round at height, confirmed previous height"
+                block_hash = hex::encode(&data.0),
+                num_topdown_messages = data.2.len(),
+                num_validator_changes = data.1.len(),
             );
-
-            let data = self.fetch_data(to_confirm_height, to_confirm_hash).await?;
-
-            atomically_or_err::<_, Error, _>(|| {
-                // we only push the null block in cache when we confirmed a block so that in cache
-                // the latest height is always a confirmed non null block.
```

Comment:
So this is no longer true.

Reply:
nope, we just need to insert whatever we pulled from parent into the cache